heudiconv/convert.py

```diff
@@ -332,18 +332,22 @@ def update_complex_name(metadata: dict[str, Any], filename: str) -> str:
 
     # Check to see if it is magnitude or phase part:
     img_type = cast(List[str], metadata.get("ImageType", []))
-    if "M" in img_type:
-        mag_or_phase = "mag"
-    elif "P" in img_type:
-        mag_or_phase = "phase"
+    if "M" in img_type or "MAGNITUDE" in img_type:
+        part = "mag"
+    elif "P" in img_type or "PHASE" in img_type:
+        part = "phase"
+    elif "REAL" in img_type:
+        part = "real"
+    elif "IMAGINARY" in img_type:
+        part = "imag"
```

Comment:
if we are approaching this , could we make it more robust right away? i.e. raise and error whenever multiple/conflicting ones are found? smth like (didn't try but sketching idea)

```python
   type_to_part = {
         "M": "mag",
         "MAGNITUDE": "mag",
         "P": "phase", 
      ...
     }
     present_types = set(type_to_part).intersection(img_type)
     if not present_types:
           raise RuntimeError(f"Data type could not be inferred from the ImageType={img_type}. Known types are: {sorted(type_to_part)}")
     elif len(present_types) == 1:
           part = type_to_part[present_types[0]]
     else:
           raise RuntimeError(f"Data type could not be inferred from the ImageType={img_type}. Multiple types matched: {present_types}")
```

or if both "M" and "MAGNITUDE" could be provided (let's assume that -- more robust):

```python
     present_parts = set(type_to_part[type] for type in img_type if type in type_to_part)
     if not present_parts:
           raise RuntimeError(f"Data type could not be inferred from the ImageType={img_type}. Known types are: {sorted(type_to_part)}")
     elif len(present_parts) == 1:
           part = present_parts[0]
     else:
           raise RuntimeError(f"Data type could not be inferred from the ImageType={img_type}. Multiple BIDS parts matched: {present_parts}")
```

Reply:
Thanks for all the help @yarikoptic and @bpinsard. I tried the second suggestion and it is indeed more robust.